src/Indicator.vala

```diff
@@ -86,6 +89,95 @@ public class Network.Indicator : Wingpanel.Indicator {
+                to show them something else if we get back
+                "Wired" from get_active_wired_name () */
+
+                string active_wired_name = get_active_wired_name ();
+
+                if (active_wired_name == _("Wired")) {
+                    display_widget.tooltip_markup = _("Connecting to wired network");
+                } else {
+                    display_widget.tooltip_markup = _("Connecting to “%s”").printf (active_wired_name);
+                }
+                break;
+            case Network.State.CONNECTING_WIFI:
+            case Network.State.CONNECTING_MOBILE:
+                display_widget.tooltip_markup = Granite.markup_accel_tooltip ({}, _("Connecting to “%s”".printf (get_active_wifi_name ())));
+                break;
+            case Network.State.CONNECTED_WIRED:
+                string active_wired_name = get_active_wired_name ();
+
+                if (active_wired_name == _("Wired")) {
+                    display_widget.tooltip_markup = _("Connected to wired network");
+                } else {
+                    display_widget.tooltip_markup = _("Connected to “%s”").printf (active_wired_name);
+                }
+                break;
+            case Network.State.CONNECTED_WIFI:
+            case Network.State.CONNECTED_WIFI_WEAK:
+            case Network.State.CONNECTED_WIFI_OK:
+            case Network.State.CONNECTED_WIFI_GOOD:
+            case Network.State.CONNECTED_WIFI_EXCELLENT:
+            case Network.State.CONNECTED_MOBILE_WEAK:
+            case Network.State.CONNECTED_MOBILE_OK:
+            case Network.State.CONNECTED_MOBILE_GOOD:
+            case Network.State.CONNECTED_MOBILE_EXCELLENT:
+                display_widget.tooltip_markup = _("Connected to “%s”").printf (get_active_wifi_name ());
+                break;
+            case Network.State.FAILED_WIRED:
+            case Network.State.FAILED_WIFI:
+            case Network.State.FAILED_VPN:
+            case Network.State.FAILED_MOBILE:
+                display_widget.tooltip_markup = _("Failed to connect");
+                break;
+            case Network.State.DISCONNECTED_WIRED:
+            case Network.State.DISCONNECTED_AIRPLANE_MODE:
+                display_widget.tooltip_markup = _("Disconnected");
+                break;
+            default:
+                display_widget.tooltip_markup = _("Not connected");
+                break;
+        }
+    }
+
+    private string get_active_wired_name () {
+        string active_wired_name = _("unknown network");
+        bool found = false;
+
+        popover_widget.other_box.get_children ().foreach ((child) => {
+            if ((child is Network.EtherInterface) && !found) {
+                found = true;
+                active_wired_name = ((Network.EtherInterface) child).active_wired_name;
+
+                debug ("Active network (Wired): %s".printf (active_wired_name));
+            }
+        });
+
+        return active_wired_name;
```

Comment:
Ah sorry, I should try pulling the code before I comment a suggestion! How about this :) 
```suggestion
        foreach (unowned Gtk.Widget child in popover_widget.other_box.get_children ()) {
            if (child is Network.EtherInterface) {
                var active_wired_name = ((Network.EtherInterface) child).active_wired_name;
                debug ("Active network (Wired): %s".printf (active_wired_name));
                return active_wired_name;
            }
        };

        return _("unknown network");
```